modules/videoio/src/cap_mjpeg_decoder.cpp

```diff
@@ -888,8 +377,9 @@ bool MotionJpegCapture::parseRiff(MjpegInputStream& in_str)
             //RiffList::m_size includes fourCC field which we have already read
             next_riff += (riff_list.m_size - 4);
 
-            AviMjpegStream mjpeg_video_stream;
-            bool is_parsed = mjpeg_video_stream.parseAvi(in_str, m_mjpeg_frames);
+            AVIReadContainer<MjpegInputStream> mjpeg_video_stream;
```

Comment:
It will be better to move _MjpegInputStream_ to the same place where _AVIReadContainer_ is located, then you will be able to avoid templates in the code.

Reply:
@mshabunin And unless the task was not to separate mjpeg codec and avi container, it turns out they will be together to one file again

Reply:
I don't see a problem here: _MjpegInputStream_ is just a convenient object for reading AVI files, it does not implement MJPEG specific behavior.

Reply:
@mshabunin I already remade, you can see